nucypher/blockchain/eth/actors.py

```diff
@@ -279,22 +274,11 @@ def connect_condition_providers(
     ) -> DefaultDict[int, List[HTTPProvider]]:
         providers = defaultdict(list)  # use list to maintain order
 
-        # check that we have endpoints for all condition chains
```

Comment:
I'm going to think on this removal a little more. 

While I can see that it is beneficial to remove it, and I like the aggressive move, there are some safety mechanisms that this code provides for any misconfigured nodes on `mainnet`; there is a separation between user-provided endpoints and public RPC endpoints. In this case user provided endpoints can either get away with not being configured, or they can be configured for any chain and circumvent any short-term preventions for using "any chain" on mainnet (of course a threshold of nodes would have to do this for it to work).

Reply:
Thought about this some more. I think we can add something like the following to keep a sanity check for user configured endpoints and remedy the aforementioned concerns:

```python
    def connect_condition_providers(
        self, user_supplied_endpoints: Dict[int, List[str]]
    ) -> DefaultDict[int, List[HTTPProvider]]:

        # check that we have mandatory user configured endpoints
        mandatory_user_configured_chains = {self.domain.eth_chain.id, self.domain.polygon_chain.id}
        if mandatory_user_configured_chains != set(user_supplied_endpoints):
            raise self.ActorError(
                f"User supplied endpoints for chains don't match mandatory chains: "
                f"{set(user_supplied_endpoints)} vs expected {mandatory_user_configured_chains}"
            )

        providers = defaultdict(list)  # use list to maintain order

        # ensure that no endpoint uri for a specific chain is repeated
        duplicated_endpoint_check = defaultdict(set)

        # User-defined endpoints for chains
        for chain_id, endpoints in user_supplied_endpoints.items():
            if not endpoints:
                raise self.ActorError(
                    f"No mandatory user endpoint for chain {chain_id}"
                )
```

- check to ensure that user-suppled chains match mandatory chains
- check to ensure that user actually supplies an endpoint for a mandatory chain
- renamed a variable because there was some name shadowing going on with `endpoints`.

wdyt?

Reply:
yep, simple :)